### ISSUE TITLE
spdx-satisfies for onlyAllow

### ISSUE THREAD

**Issue by DeTeam** [davglass/license-checker#164]
At the moment `onlyAllow` option would perform process.exit on first occurrence of a license which is not in the list. If I'm not mistaken, the match has to be exact, which makes one specify all possible version os apache or bsd licenses.

I'm wondering if it'd be better to soften the way it works here and use `spdxSatisfies` as it's done for `exclude` option.

@davglass

**Comment by davglass** [comment]
Yes, this needs to be handled. Ideally all license checking should be via `spdxSatisfies`, I've just had no time to look into it 😞

**Comment by DeTeam** [comment]
I'm thinking about the best way to contribute given this and another issue I created, @davglass
In addition to these two I feel like it might be nice to provide a more reach/modular js api (I could also adjust TS definitions for it)  so that one can have granular control over how license checking works. What do you think?
